doc/source/ray-core/examples/batch_prediction.ipynb

```diff
@@ -0,0 +1,283 @@
+    "    pool.submit(lambda a, v: a.predict.remote(v), file)\n",
+    "while pool.has_next():\n",
+    "    print(\"r size:\", pool.get_next())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Batch prediction with GPUs\n",
+    "\n",
+    "If your cluster has GPU nodes and your predictor can utilize the GPUs, you can direct the tasks or actors to those GPU nodes by specifying num_gpus. Ray will schedule them onto GPU nodes accordingly."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@ray.remote # Replace with @ray.remote(num_gpus=1) to use GPUs\n",
+    "def make_prediction(model, shard_path):\n",
+    "    df = pq.read_table(shard_path).to_pandas()\n",
+    "    result = model(df)\n",
+    "    # Write the result out.\n",
+    "    # Here we just return the size about the result.\n",
+    "    return len(result)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## FAQ\n",
+    "\n",
+    "### How to load and pass model efficiently in Ray cluster if the model is large?\n",
+    "The recommended way is to (taking task-based batch prediction for example, the actor-based is the same):\n",
+    "1. let the driver load the model (e.g. from storage system)\n",
+    "2. let the driver ray.put(model) to store the model into object store; and\n",
+    "3. pass the same reference of the model to each remote tasks when launching them.\n",
+    "The remote task will fetch the model (from driver's object store) to its local object store before start performing prediction.\n",
+    "\n",
+    "Note it's highly inefficient if you skip the step 2 and pass the model (instead of reference) to remote tasks. If the model is large and there are many tasks, it'll likely cause out-of-disk crash for the driver node."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Good: the model will be stored to driver's object store only once\n",
+    "model = load_model()\n",
+    "model_ref = ray.put(model)\n",
+    "for file in input_files:\n",
+    "    make_prediction.remote(model_ref, file)\n",
+    "\n",
+    "# Bad: the same model will be stored to driver's object store repeatedly for each task\n",
+    "model = load_model()\n",
+    "for file in input_files:\n",
+    "    make_prediction.remote(model, file)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### How to improve the GPU utilization rate?\n",
+    "To keep GPUs busy, there are following things to look at:\n",
+    "- **Make sure GPU is not starving on data**: If the time-to-deliver data to GPU is longer than time-to-consume data by GPU, you may want to optimize the data deliver. A common technique is data prefetching: fetch next batch of data while the current one is still being consumed by GPU.\n",
```

Comment:
We need code examples otherwise I think most people don't know how to do it.

Reply:
What about this comment?

Reply:
What's the design pattern here to run prefetch in core?

Reply:
I don't think we have an existing design pattern for it.

I think the way to do that is submitting more tasks than gpus so while some tasks are running, Ray can fetch args for remaining tasks in the background.

Reply:
It's a bit different here : the prefetch needed here is to read storage system (not prefetch remote Ray objects). It looks there is no concrete guide from us, I'm taking this item down since it's very actionable for now.